src/vs/workbench/api/common/extHostTerminalService.ts

```diff
@@ -258,12 +258,45 @@ export class ExtHostTerminal extends Disposable {
 
 	public setInteractedWith(): boolean {
 		if (!this._state.isInteractedWith) {
-			this._state = { isInteractedWith: true };
+			this._state = {
+				...this._state,
+				isInteractedWith: true
+			};
 			return true;
 		}
 		return false;
 	}
 
+	public setShellType(shellType: TerminalShellType | number | undefined): boolean {
+		let extHostType: vscode.TerminalShellType | undefined;
+		// Problem: Turns out shellType is a string, so we need to map it..
+
+		switch (shellType) {
+			case 'sh': extHostType = 1; break;
+			case 'bash': extHostType = 2; break;
+			case 'fish': extHostType = 3; break;
+			case 'csh': extHostType = 4; break;
+			case 'ksh': extHostType = 5; break;
+			case 'zsh': extHostType = 6; break;
+			case 'cmd': extHostType = 7; break;
+			case 'gitbash': extHostType = 8; break;
+			case 'pwsh': extHostType = 9; break;
+			case 'python': extHostType = 10; break;
+			case 'julia': extHostType = 11; break;
+			case 'nu': extHostType = 12; break;
+			// TODO: default? as undefined?
```

Comment:
Yes default is undefined, which we can set or not (since extHostType is already undefined). we should also remove `| number` from setting since that won't even happen, if we add a new one it'll be part of `TerminalShellType`.

Reply:
I removed ` | number ` from everywhere except here:
```
	export interface TerminalState {
		/**
		 * The current detected shell type of the terminal. New shell types may be added in the
		 * future in which case they will be returned as a number that is not part of
		 * {@link TerminalShellType}.
		 * TODO: number is to prevent the breaking change when new enum members are added?
		 */
		readonly shellType?: TerminalShellType | number | undefined;
	}
```